fdbserver/TLogServer.actor.cpp

```diff
@@ -2858,6 +2865,10 @@ ACTOR Future<Void> serveTLogInterface(TLogData* self,
 		when(TLogEnablePopRequest enablePopReq = waitNext(tli.enablePopRequest.getFuture())) {
 			logData->addActor.send(tLogEnablePopReq(enablePopReq, self, logData));
 		}
+		when(SendClusterRecoveryVersionRequest req = waitNext(tli.sendClusterRecoveryVersion.getFuture())) {
+			self->clusterRecoveryVersion = req.recoveryVersion;
+			req.reply.send(Void());
+		}
```

Comment:
The cluster controller is sending this message, right?

Reply:
It appears cluster recovery version is the recovery version and you are setting it unconditionally, so I am confused by your comment above that it is set based on "reqBegin".

Reply:
And another comment regarding cluster recovery version: it appears we never reset it on a locked log, is that going to be a problem if another recovery starts before the current recovery completes?

Reply:
> The cluster controller is sending this message, right?

see PR comment "The message will be sent from epochEnd() in a later iteration of the PR." Once your PR is merged I'll update the PR and run them together

>It appears cluster recovery version is the recovery version and you are setting it unconditionally, so I am confused by your comment above that it is set based on "reqBegin".

discussed offline about line 1841

>And another comment regarding cluster recovery version: it appears we never reset it on a locked log, is that going to be a problem if another recovery starts before the current recovery completes?

I don't think so, because we would resend the RV if it is recalculated in `epochEnd()`. But we only call `pullAsync`  once we start the new epoch, and at that point the RV is fixed and any new recovery would be a new epoch. That said we will need to test to be fully confident.